src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -261,6 +261,39 @@ public void testBAMThroughCRAMRoundTrip() throws IOException, NoSuchAlgorithmExc
         }
     }
 
+    @Test
+    public void testBAMThroughCRAMRoundTripViaPath() throws IOException, NoSuchAlgorithmException {
+        final File TEST_DATA_DIR = new File("src/test/resources/htsjdk/samtools/cram");
+
+        // These files are reduced versions of the CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam and human_g1k_v37.20.21.fasta
+        // files used in GATK4 tests. The first 8000 records from chr20 were extracted; from those around 80 placed but
+        // unmapped reads that contained cigar elements were removed, along with one read who's mate was on chr21.
+        // Finally all read positions were remapped to the subsetted reference file, which contains only the ~9000 bases
+        // used by the reduced read set.
+        final File originalBAMInputFile = new File(TEST_DATA_DIR, "CEUTrio.HiSeq.WGS.b37.NA12878.20.first.8000.bam");
+        final File referenceFile = new File(TEST_DATA_DIR, "human_g1k_v37.20.subset.fasta");
+
+        // retrieve all records from the bam and reset the indexing bins to keep comparisons with
+        // cram records from failing
+        List<SAMRecord> originalBAMRecords = getSAMRecordsFromFile(originalBAMInputFile, referenceFile);
+        for (int i = 0; i < originalBAMRecords.size(); i++) {
+            originalBAMRecords.get(i).setIndexingBin(null);
+        }
+
+        // write the BAM records to a temporary CRAM
+        final File tempCRAMFile = File.createTempFile("testBAMThroughCRAMRoundTrip", CramIO.CRAM_FILE_EXTENSION);
```

Comment:
Excellent idea! I see we already depend on jimfs.

I'm also adding it to the test of `IOUtil.addExtension`, so we can test non-default-filesystem paths.

I also changed testCRAMThroughBAMRoundTrip so it saves to jimfs instead of the filesystem, that should speed things up a little bit.

This leaves only one test using files. I'm thinking of leaving it so we exercise that path through the code, though we can remove it if you feel strongly about it (these methods just call `.toPath()` and then redirect to the path code anyways).